app/scripts/migrations/069.js

```diff
@@ -0,0 +1,40 @@
+import { cloneDeep } from 'lodash';
+import { SUBJECT_TYPES } from '../../../shared/constants/app';
+
+const version = 69;
+
+/**
+ * Adds the `subjectType` property to all subject metadata.
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const { SubjectMetadataController = {} } = state;
+  const { subjectMetadata = {} } = SubjectMetadataController;
+
+  // mutate SubjectMetadataController.subjectMetadata in place
+  Object.values(subjectMetadata).forEach((metadata) => {
+    if (metadata && typeof metadata === 'object' && !Array.isArray(metadata)) {
```

Comment:
Nit: These branches seem untested. You could add a test with malformed subject metadata (e.g. the metadata is an array, or is something other than an object).

And if we want to protect against malformed state in general, it might be worth ensuring `subjectMetadata` is an object before invoking `Object.values` with it.